packages/lib-utils/src/components/list-view/ListView.test.tsx

```diff
@@ -0,0 +1,208 @@
+import { screen, render, fireEvent } from '@testing-library/react';
+import * as React from 'react';
+import { act } from 'react-dom/test-utils';
+import { MemoryRouter } from 'react-router-dom';
+import type { TableTestItem } from '../table/test-data';
+import {
+  testData as data,
+  TestRow as Row,
+  testColumns as columns,
+  testFilters,
+} from '../table/test-data';
+import ListView, { filterDefault } from './ListView';
+
+describe('TableView - non-virtualized', () => {
+  it('should render with data', () => {
+    render(
+      <MemoryRouter initialEntries={[{ pathname: '/hac' }]}>
+        <ListView
+          data={data}
+          columns={columns}
+          Row={Row}
+          filters={testFilters}
+          loaded
+          isRowSelected={undefined}
+          onSelect={undefined}
+          onFilter={undefined}
+          loadError={undefined}
+          CustomEmptyState={undefined}
+          emptyStateDescription={undefined}
+          CustomNoDataEmptyState={undefined}
+          aria-label={undefined}
+        />
+      </MemoryRouter>,
+    );
+    data.forEach((item, index) => {
+      expect(screen.getByTestId(`col-name-${index}`).textContent).toEqual(item.name);
+      expect(screen.getByTestId(`col-prs-${index}`).textContent).toEqual(item.prs);
+      expect(screen.getByTestId(`col-workspaces-${index}`).textContent).toEqual(
+        String(item.workspaces),
+      );
+      expect(screen.getByTestId(`col-branches-${index}`).textContent).toEqual(item.branches);
+    });
+  });
+
+  it('should show/hide chips & clear all button on search/clear', async () => {
+    jest.useFakeTimers();
+    await act(async () => {
+      render(
+        <MemoryRouter initialEntries={[{ pathname: '/hac' }]}>
+          <ListView
+            data={data}
+            columns={columns}
+            Row={Row}
+            filters={testFilters}
+            loaded
+            isRowSelected={undefined}
```

Comment:
Not blocking - I'd like for us to investigate de-duplicating these props across our tests (at least the ones that remain consistent between tests). Otherwise it's very hard to identify which props are changing between tests (if any at all)

Reply:
Yes, and I'm not sure if we really need explicit `someProp={undefined}` ones as well, can be looked into later on.